lib/lightstep/tracer/client_tracer.rb

```diff
@@ -1,86 +1,83 @@
-require 'securerandom'
 require 'json'
 
 require_relative './client_span'
-require_relative './no_op_span'
 require_relative './util'
 require_relative './transports/transport_http_json'
 require_relative './thrift/types'
 require_relative './version.rb'
 
-
 # ============================================================
- # Main implementation of the Tracer interface
+# Main implementation of the Tracer interface
 # =============================================================
 
 class ClientTracer
-
   def initialize(options = {})
-
     @tracer_utils = Util.new
     @tracer_options = {}
     @tracer_enabled = true
     @tracer_debug = false
-    @tracer_guid = ""
+    @tracer_guid = ''
     @tracer_start_time = @tracer_utils.now_micros
     @tracer_thrift_auth = nil
     @tracer_thrift_runtime = nil
     @tracer_transport = nil
     @tracer_report_start_time = 0
-    @tracer_log_records = Array.new
-    @tracer_span_records = Array.new
-    @tracer_counters = {:dropped_logs => 0, :dropped_counters => 0}
+    @tracer_log_records = []
+    @tracer_span_records = []
+    @tracer_counters = {
+      dropped_logs: 0,
+      dropped_spans: 0
+    }
     @tracer_last_flush_micros = 0
-    @tracer_min_flush_period_micros = 0
-    @tracer_max_flush_period_micros = 0
+    @tracer_min_flush_period_micros = 500 * 1000
+    @tracer_max_flush_period_micros = 30_000 * 1000
```

Comment:
Lines 32-33 are the key to the fix: they were previously initialized to 0 which meant the code thought it needed to constantly flush the buffers.